### DISCUSSION TITLE
Discuss: Support Conventionnal Comments

### DISCUSSION THREAD

**Discussion by silopolis** [alstr/todo-to-issue-action#157] [Ideas]
Hi,

First, thanks for this rocking project. I'm a huge fan of TODOs, FIXMEs and other "typed comments", after IDEs/editors extensions supporting them, integration with code management platform just puts them on steroid!

I'm opening this more for discussion than to request features, but don't hold yourself ;)

Once more serious maters will be addressed, it could be fantastic to leverage those efforts by supporting [Conventional Comments](https://conventionalcomments.org/) ([repository](https://gitlab.com/conventionalcomments)). This could create potential uses in so many different cases!

See [this issue](https://gitlab.com/conventionalcomments/conventionalcomments.gitlab.io/-/issues/20) about the definition of a format spec.

TY

**Comment by alstr** [comment]
Thanks for the positive comments. I'm always open to improving the action and for others to chip in however they want.

How do you propose the Conventional Comments support could work? Sounds interesting.

**Comment by silopolis** [comment]
After spending some time on this matter, and reading again the README (hence #145), I finally believe that the specification to follow should be the [Conventionnal Commits'](https://www.conventionalcommits.org/en/v1.0.0/) one (itself integrating the [git trailer convention](https://git-scm.com/docs/git-interpret-trailers)), already leveraged by the [Conventional Changelog](https://github.com/conventional-changelog/conventional-changelog-config-spec) one and a growing set of tools.

Good thing is that there seems to be no big difference with the current implementation and that it would work pretty much as it does today :)

I believe there could be nice opportunities in using the same, or compatible, standard from the initial code comment (block), to commit, and review. These could emerge both in-process for automation, management, and communication, as well as for code/project documentation to extend or complement changelog/release notes. Could be another powerful tool in the #docs-as-code chest...

**Comment by alstr** [comment]
The latest update does include some new TODO reference options, e.g. `TODO(!mylabel)`. That is about as close as I've gotten to supporting Conventional Commits at the moment, but I'm always open to further ideas.
